user/reference/xenial.md

```diff
@@ -0,0 +1,153 @@
+
+```yaml
+dist: xenial
+addons:
+  apt:
+    sources:
+      - ppa:chris-lea/redis-server
+    packages:
+    - redis-tools
+    - redis-server
+```
+{: data-file=".travis.yml"}
+
+### Third party packages available in Xenial
+
+The following table outlines the third party packages available in Xenial, along with their original source.
+
+| package       | source                                                                      |
+|:--------------|:----------------------------------------------------------------------------|
+| redis         | `ppa:chris-lea/redis-server`                                                |
+| haskell       | `ppa:hvr/ghc`                                                               |
+| mongodb       | `"deb http://repo.mongodb.org/apt/ubuntu xenial/mogodb-org/4.0 multiverse"` |
+| couchdb       | `ppa:couchdb/stable`                                                        |
+| docker        | `"deb [arch=amd64] https://download.docker.com/linux/ubuntu xenial stable"` |
+| git-lfs       | `"deb https://packagecloud.io/github/git-lfs/ubuntu/ xenial main"`          |
+| git-ppa       | `ppa:git-core/ppa`                                                          |
+| google_chrome | `"deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main"`     |
+| pollinate     | `ppa:pollinate/ppa`                                                         |
+
+### Services disabled by default
+
+On Trusty, some services that you might or might not use, were running by
+default. On Xenial, to optimize the use of the virtual machine resources, they come disabled by default.
+
+You can start them by specifying the services that you require. For example:
+
+```yaml
+services:
+  - mysql
+  - redis
+```
+{: data-file=".travis.yml"}
+
+## Environment common to all Xenial images
+
+In total, three different Xenial images are available for use. The selection
+process takes place based on the language and services specified in your `travis.yml` configuration.
+
+The minimal image `stevonnie` supports docker, bash, and has the tools for dynamic language runtime download and activation, for example, `rvm` for ruby.
+
+The other two images have different language runtimes installed by default, and
+have an array of different databases and services available.
+
+You can find below the environment common to all Xenial images. The environment specifications below are also the ones for our `language: minimal` Xenial image.
+
+### Version control
+
+| package | version  |
+|:--------|:---------|
+| git     | `2.19.1` |
+| git-lfs | `2.6.0`  |
+| hg      | `4.8`    |
+| svn     | `1.9.3`  |
+{: style="width: 50%" } 
+
+### Compilers and Build toolchain
+
+* clang and llvm 7 are installed.
+* cmake 3.12.4 is available. The gcc toolchain is available from the Ubuntu
+repositories.
+
+### Docker
+
+* Docker 18.06.0-ce is installed
+* docker-compose 1.23.1.
+
+## Ruby images
+
+* Pre-installed Rubies: `2.4.5` and `2.5.3`.
```

Comment:
What we could do is add a line for each language here about what base image is uses?